docs/aws/audit/lambdamonitoring/rules/check_cross_account_access_remediation.mdx

```diff
@@ -1,6 +1,206 @@
+   if __name__ == "__main__":
+       functions = list_lambda_functions()
+       for function in functions:
+           remove_cross_account_access(function['FunctionName'])
+   ```
+
+By following these steps, you can ensure that your Lambda functions do not allow cross-account access, thereby enhancing the security of your AWS environment.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the AWS Lambda service by typing 'Lambda' into the search bar and selecting 'Lambda' from the dropdown menu.
+3. In the AWS Lambda console, select the 'Functions' option from the left-hand navigation pane. This will display a list of all the Lambda functions in your account.
+4. For each Lambda function, select the function to view its details. Under the 'Permissions' tab, check the 'Resource-based policy' section. If there are any statements with "Principal": "*" (which allows cross-account access) or `"Principal": {"AWS": "arn:aws:iam::<account-id>:root"}` (which allows access from a specific AWS account), then the Lambda function is misconfigured to allow cross-account access.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the Lambda functions in your AWS account. You can do this by using the AWS CLI command `aws lambda list-functions`. This command will return a list of all the Lambda functions in your AWS account.
+
+2. For each Lambda function, you need to get the resource policy. You can do this by using the AWS CLI command `aws lambda get-policy --function-name <function-name>`. Replace `<function-name>` with the name of your Lambda function. This command will return the resource policy of the Lambda function.
+
+3. Once you have the resource policy, you need to check if there are any statements that allow cross-account access. You can do this by looking for any statements where the `Principal` field is not equal to your AWS account ID or `*`.
+
+4. If you find any such statements, then the Lambda function allows cross-account access. If you don't find any such statements, then the Lambda function does not allow cross-account access. 
+
+Note: You can automate these steps by writing a Python script that uses the AWS SDK (boto3) to interact with the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary AWS SDK for Python (Boto3) modules in your Python script. You will need the 'boto3' module to interact with AWS services and the 'json' module to parse the AWS Lambda function policy.
+
+```python
+import boto3
+import json
+```
+
+2. Create a session using your AWS credentials. Replace 'aws_access_key_id', 'aws_secret_access_key', and 'aws_session_token' with your actual AWS credentials.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>